daphne/http_protocol.py

```diff
@@ -232,7 +232,7 @@ def __init__(self, channel_layer, action_logger=None, timeout=120, websocket_tim
         # We track all sub-protocols for response channel mapping
         self.reply_protocols = {}
         # Make a factory for WebSocket protocols
-        self.ws_factory = WebSocketFactory(self)
+        self.ws_factory = WebSocketFactory(self, protocols=ws_protocols)
```

Comment:
the `protocols` argument is already inherited from the signature in `autobahn.websocket.protocol.WebSocketServerFactory`